docs/proposals/pod-troubleshooting.md

```diff
@@ -0,0 +1,573 @@
+insufficient for tasks like modifying volumes and their mount points. For more
+complex changes, we can provide an `--edit` workflow to adjust the generated
+spec before creation:
+
+```
+# Similar to the following workflow:
+#   kubectl get -o yaml ... > $temp_file
+#   $EDITOR $temp_file
+#   kubectl create -f $temp_file
+kubectl run --copy-of=target-pod -it -c shell --image=debian --edit
+```
+
+Note that the `--edit` workflow is useful for all generated config, not just
+copies of pods.
+
+#### Scheduling conflicts
+
+Attempting to duplicate some resources will cause the resulting pod to be
+unschedulable, for example creating a second read-write volume mount of a
+gcePersistentDisk will create a disk conflict. This is the same problem faced by
+scaling a deployment, so one way to provide a better user experience is to
+validate the generated config similar to `ValidatePodTemplateSpecForRC()` with
+replicas > 1 as `kubectl scale` does.
+
+Validating client side is not optimal, and currently only gcePersistentDisk
+provides a check in `ValidatePodTemplateSpecForRC()`, but this isn't really a
+new problem introduced by this new functionality.
+
+#### Stripping labels
+
+Only the spec will be copied from the running pod. Copying metadata such as
+labels might result in the pod copy receiving traffic or being killed
+immediately by a replication controller. It would be trivial to provide a
+`--copy-labels` option if this is desired behavior.
+
+### Running Debug
+
+*Running Debug* requires new functionality in the kubelet and a new `kubectl
+debug` command that results in a new container being introduced in the running
+pod. This *debug container* is not part of the pod spec, which remains
+immutable, and is not restarted according to the pod's `restartPolicy`.
+
+The status of a *debug container* is reported in a new `DebugContainerStatuses`
+field of `PodStatus`, which is a read-only list of `ContainerStatus` that
+contains an entry for every *debug container* that has ever run in this pod, and
+is reported by `kubectl describe`. This mirrors how `InitContainers` are
+implemented.
+
+Debug operations will generate an event so auditing utilities can reconstruct
+what commands are run, even though `kubectl exec` does not do this. A *Debug
+Containers* section will display all debug containers in the output of `kubectl
+describe` similar to the *Init Containers* section.
+
+The following command would attach to a newly created container in a pod:
+
+```
+kubectl debug -it -c debug-shell --image=debian target-pod -- bash
+```
+
+It would be reasonable for Kubernetes to provide a default container image
+which, combined with a default entrypoint, makes the minimal debug command:
+
+```
+kubectl debug -p target-pod -it
+```
+
+If the specified container name already exists, `kubectl debug` will kill the
+existing container prior to starting a new one. It's legal to reuse container
```

Comment:
I think asking the user before replacing it is more reasonable. 

Another thing is that the container names cannot be reused as long as the container still exists, even if it already exited. You'll need to remove (GC) the container, which may cause the loss of one "DebugContainerStatus".